### PR TITLE
Fixed makefile for unit tests.

### PR DESCRIPTION
**Pull request by martinbudden**

### CODE REVIEW

**Comment by hydra** [comment]
@martinbudden build failed.

**Comment by martinbudden** [comment]
@hydra,

the error is a linker error. It builds fine on my machine (a Mac), so the error is probably something to do with the parameter groups, i.e.
```
# Determine the OS and set up the parameter group linker flags accordingly
UNAME := $(shell uname -s)
ifeq ($(UNAME), Darwin)
PG_FLAGS = -Wl,-map,$(OBJECT_DIR)/$@.map
else
PG_FLAGS = -Wl,-T,$(TEST_DIR)/parameter_group.ld -Wl,-Map,$(OBJECT_DIR)/$@.map
endif
```
This means I can't really troubleshoot this any further.

@ledvinap any ideas?

**Comment by martinbudden** [comment]
@hydra,

although this doesn't fix all the problems, it nevertheless should be merged in, since it fixes some of them.

File: src/main/common/utils.h [hunk 1]
```diff
@@ -52,6 +52,10 @@ static inline int32_t cmp32(uint32_t a, uint32_t b) { return a-b; }
 
 // using memcpy_fn will force memcpy function call, instead of inlining it. In most cases function call takes fewer instructions
 //  than inlined version (inlining is cheaper for very small moves < 8 bytes / 2 store instructions)
+#ifdef UNIT_TEST
```
Comment (ledvinap) [inline src/main/common/utils.h]: Is this really really necessary? Both gcc and clang seems to handle this correctly ...

File: src/test/Makefile [hunk 2]
```diff
@@ -31,7 +31,6 @@ endif
 
 COMMON_FLAGS = \
 	-g \
-	-pthread \
```
Comment (ledvinap) [inline src/test/Makefile]: `-pthread` is ok, only arm-none-eabi-gcc did not understand it

File: src/test/Makefile [hunk 3]
```diff
@@ -51,8 +50,8 @@ CXX_FLAGS = $(COMMON_FLAGS) \
 GCOV_CC_FLAGS := -fprofile-arcs -ftest-coverage
 GCOV_L_FLAGS  := -lgcov
```
Comment (ledvinap) [inline src/test/Makefile]: `-fprofile-arcs` implies `-lgcov` when linking, so `lgcov` can be safely removed

And even better would be `--coverage` argument - it is supported both in clang and gcc and enables -fprofile-arcs -ftest-coverage/ -lgcov as necessary ..

**Comment by martinbudden** [comment]
@hydra , @ledvinap this is now working with `FAIL_ON_WARNINGS` set.

**Comment by ledvinap** [comment]
@martinbudden : The idea was to set FAIL_ON_WARNINGS only when building PR, but build normally when building CF branches..
Moving PR detection code from Makefile to .travis.sh is probably best option ...

**Comment by ledvinap** [comment]
@martinbudden : https://github.com/cleanflight/cleanflight/pull/2194

**Comment by martinbudden** [comment]
@ledvinap , OK, I'll withdraw this PR in favour of yours.

**Pull request closed without merging**